src/kakarot/accounts/generic_account.cairo

```diff
@@ -0,0 +1,217 @@
+// @return response_len The length of the response array
+// @return response The response from the kakarot contract
+@external
+func __execute__{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    ecdsa_ptr: SignatureBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    range_check_ptr,
+}(
+    call_array_len: felt, call_array: GenericAccount.CallArray*, calldata_len: felt, calldata: felt*
+) -> (response_len: felt, response: felt*) {
+    alloc_locals;
+    let (tx_info) = get_tx_info();
+    let version = tx_info.version;
+    with_attr error_message("EOA: deprecated tx version: {version}") {
+        assert_le(1, version);
+    }
+
+    let (caller) = get_caller_address();
+    with_attr error_message("EOA: reentrant call") {
+        assert caller = 0;
+    }
+
+    let (local response: felt*) = alloc();
+    let (response_len) = GenericAccount.execute(
+        call_array_len=call_array_len,
+        call_array=call_array,
+        calldata_len=calldata_len,
+        calldata=calldata,
+        response=response,
+    );
+    return (response_len, response);
+}
+
+// @dev Return true if the interface_id is supported
+// @dev TODO: check what interfaces the contract should support and maybe create one for a kakarot account
+// @param interface_id The interface Id to verify if supported
+@view
+func supports_interface{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    interface_id: felt
+) -> (success: felt) {
+    if (interface_id == GenericAccount.INTERFACE_ID) {
+        return (success=1);
+    }
+    return (success=0);
+}
+
+// @notice Store the bytecode of the contract.
+// @param bytecode The bytecode of the contract.
+// @param bytecode_len The length of the bytecode.
+@external
+func write_bytecode{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}(bytecode_len: felt, bytecode: felt*) {
+    return GenericAccount.write_bytecode(bytecode_len, bytecode);
+}
+
+// @notice This function is used to get the bytecode of the smart contract.
+// @return bytecode_len The bytecode array length.
+// @return bytecode The bytecode of the smart contract.
+@view
+func bytecode{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}() -> (bytecode_len: felt, bytecode: felt*) {
+    return GenericAccount.bytecode();
+}
+
+// @notice This function is used to get only the bytecode_len of the smart contract.
+// @dev Compared to bytecode, it does not read the code so it's much cheaper if only len is required.
+// @return len The bytecode_len of the smart contract.
+@view
+func bytecode_len{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}() -> (len: felt) {
+    let (len) = GenericAccount.bytecode_len();
+    return (len=len);
+}
+
+// @notice Store a key-value pair.
+// @param key The storage address, with storage_var being storage_(key: Uint256)
```

Comment:
so actualluy the `Account_storage` is not used directly but was left for hinting that the storage, though beging writen directly with the address (as it was easier from the main Kakarot contract) actually is this one.
So you should update the comment, and the hard coded constant in Kakarot account.cairo

If you think that this is unclear, we can think about another way.

Reply:
ah yeah I missed that comment, but correctly updated _storage_addr